pineappl_capi/include/PineAPPL.hpp

```diff
@@ -119,15 +115,79 @@ struct Lumi {
   }
 
   /**
-   * @brief Returns the number of combinations of the luminosity function `lumi`
-   * for the specified entry.
+   * @brief Returns the number of combinations of the luminosity function
+   * `lumi` for the specified entry.
    * @param entry position in lumi
    */
   std::size_t combinations(const std::size_t entry) const {
     return pineappl_lumi_combinations(this->raw, entry);
   }
 };
 
+/** @brief Entry in sub-channel function.  */
+struct SubChannelEntry {
+  /** @brief First parton id. */
```

Comment:
1. since this is actually complicated we need some documentation here
2. this is a case of "inheritance vs. composition", not sure which one is better ... however, are there any additional methods or attributes in the first place? then also a simple `typedef` might be sufficient

Reply:
> this is a case of "inheritance vs. composition", not sure which one is better ... however, are there any additional methods or attributes in the first place? then also a simple typedef might be sufficient

Yes, `SubChannelEntry` has/can have additional methods/attributes (such as returning the combination of channels for a given entry) which one might want to expose in the future, so I'd say a generalized struct is better in this case.

Btw, the "main" interfaces should now be complete if you want to have a look/improve.